manager/controlapi/secret.go

```diff
@@ -71,7 +72,10 @@ func (s *Server) UpdateSecret(ctx context.Context, request *api.UpdateSecretRequ
 			return nil
 		}
 
-		if secret.Spec.Annotations.Name != request.Spec.Annotations.Name || request.Spec.Data != nil {
+		// Check if the Name is different than the current name, or the secret is non-nil and different
+		// than the current secret
+		if secret.Spec.Annotations.Name != request.Spec.Annotations.Name ||
+			(request.Spec.Data != nil && !bytes.Equal(request.Spec.Data, secret.Spec.Data)) {
```

Comment:
I guess it's okay. I'm a little worried that we'll get into ambiguous cases eventually, like with enums. ("Is the client trying to set this value to the default, or is it just not specifying it?")

Since the secret payload is sensitive, should we use `subtle.ConstantTimeCompare`?